This patch fixes the build failure caused by the FLUSH macro,
which is defined in /usr/include/sys/controlregs.h file.
The change undefines FLUSH macro in DrawCommand.h file
before the enum CommandType with FLUSH item is created.
Bug 28892813 was created to indicate that the name of
the global define in controlregs.h file (system/header
component) is too general.

The second part fixes the build failure for psutils
as described in https://github.com/giampaolo/psutil/pull/1003.
Macros MA_NORESERVE and MA_RESERVED1 are not currently
used and defined in Solaris, therefore, they can be
safely removed.

The FLUSH macro is defined in Solaris specific header file,
therefore, the undef workaround will not be sent to upstream.
The fixes for psutils are already in upstream repository.

--- firefox-52.9.0esr/gfx/2d/DrawCommand.h	2018-11-06 08:54:05.385195458 +0000
+++ firefox-52.9.0esr/gfx/2d/DrawCommand.h	2018-11-06 08:52:58.163990889 +0000
@@ -6,6 +6,8 @@
 #ifndef MOZILLA_GFX_DRAWCOMMAND_H_
 #define MOZILLA_GFX_DRAWCOMMAND_H_
 
+#undef FLUSH
+
 #include <math.h>
 
 #include "2D.h"
--- firefox-52.9.0esr/python/psutil/psutil/_psutil_sunos.c	2018-11-06 08:51:50.960869143 +0000
+++ firefox-52.9.0esr/python/psutil/psutil/_psutil_sunos.c	2018-11-06 08:50:28.757002411 +0000
@@ -616,9 +616,7 @@ psutil_proc_memory_maps(PyObject *self,
         sprintf(perms, "%c%c%c%c%c%c", p->pr_mflags & MA_READ ? 'r' : '-',
                 p->pr_mflags & MA_WRITE ? 'w' : '-',
                 p->pr_mflags & MA_EXEC ? 'x' : '-',
-                p->pr_mflags & MA_SHARED ? 's' : '-',
-                p->pr_mflags & MA_NORESERVE ? 'R' : '-',
-                p->pr_mflags & MA_RESERVED1 ? '*' : ' ');
+                p->pr_mflags & MA_SHARED ? 's' : '-');
 
         // name
         if (strlen(p->pr_mapname) > 0) {
